Activity to introduce tutorial interface

kiki-lee opened the pull request:
Making a PR to get suggestions.  There's still some roughness I'd like help smoothing out:

1) Made a new .gif that needs to be uploaded (once it clears PR) to replace the blocks with text on step 4.
2) I can't get the effects blocks to show correctly.  I tried setting sprites in ghost blocks and a few other things, but it just stays grey. (Step 8)
3) Can't figure out how to get parameters rendered outside of a larger block. (Step 9)
4) Also in step 9, the 'mySprite say' block is grey.  I think both 2 & 4 have to do with 'mySprite' not being defined as part of the main code, but I can't figure out a good way to do that.  I don't want to add it in the template, and the ghost code didn't seem to be working the way it's documented.
5) I want other blocks available in the toolbox, but can't figure out how to get them added without the ghost feature.


(2)
![image](https://user-images.githubusercontent.com/1633958/101594054-e6d46b80-39a5-11eb-8c71-02ffa83908c6.png)

(3)
![image](https://user-images.githubusercontent.com/1633958/101594075-ec31b600-39a5-11eb-8cbb-374234b81ac7.png)

kiki-lee committed:
Activity to introduce tutorial interface

Making a PR to get suggestions.  There's still some roughness I'd like help smoothing out:

1) Made a new .gif that needs to be uploaded (once it clears PR) to replace the blocks with text on step 4.
2) I can't get the effects blocks to show correctly.  I tried setting sprites in ghost blocks and a few other things, but it just stays grey. (Step 8)
3) Can't figure out how to get parameters rendered outside of a larger block. (Step 9)
4) Also in step 9, the 'mySprite say' block is grey.  I think both 2 & 4 have to do with 'mySprite' not being defined as part of the main code, but I can't figure out a good way to do that.  I don't want to add it in the template, and the ghost code didn't seem to be working the way it's documented.
5) I want other blocks available in the toolbox, but can't figure out how to get them added without the ghost feature.

pelikhan commented:
@shakao @riknoll is there an automated build for this? can we have a github action to build PRs?

pelikhan commented on tutorials/interface/activity1.md:
```diff
@@ -0,0 +1,242 @@
+
+## Introduction @unplugged
+
+![Psyched Monkey](https://github.com/microsoft/pxt-arcade/blob/master/docs/static/tutorials/monkey.png?raw=true "Psyched Monkey is Ready!" )
```
@shakao  @riknoll what's the story for image in skill map context? Pulling files from github won't work.

shakao replied:
hmm haven't thought too much about this but if you publish on github pages it hosts all your content (that's how we do the the certificates right now), so something like `https://microsoft.github.io/pxt-skillmap-sample/img/space/activity1.png` would work for the images also

pelikhan replied:
If github pages are enabled, this will work

https://microsoft.github.io/pxt-skillmap-sample/img/platformer/activity2.png

kiki-lee replied:
Talked to shakao on this and should be able to use a relative link if I keep putting images in the pxt-arcade tutorials folder.

pelikhan commented:
@shakao @riknoll how do we test a PR? it is possible to load a skillmap from a branch. This would be very helpful.

riknoll commented on tutorials/interface/activity1.md:
````diff
@@ -0,0 +1,242 @@
+## Standard Blocks
+
+```ghost
+mySprite.startEffect(effects.spray)
````
you need to declare mySprite like this:

```
let mySprite: Sprite = null;
```

same with the other snippet. All snippets need to be working bits of code so variables always need to be declared

shakao commented:
2. For step 8, you need `let mySprite: Sprite = null` before you call `startEffect` on mySprite. You're totally right about why it's happening, which is that "mySprite" is not defined and the compiler needs the type information to generate the blocks!

3. Unfortunately we don't support floating argument blocks at the moment.

4-5. Same as in step 8, I think if you add the `let mySprite: Sprite = null` at the top of the code block, you should be able to see the blocks render and use ghost to add the categories you want to the toolbox!

kiki-lee committed:
Update activity1.md

Everything should be fixed except the tutorial blocks.  I have an image to upload for it, but paste/drop isn't working in editor.

kiki-lee commented:
This should be ready to go except for the imagery, but I'll add that properly once this is merged and the correct folder is created.

pelikhan commented:
@shakao @riknoll what is the command to compile the content and check that all sample don't have typos?

pelikhan commented on tutorials/interface/activity1.md:
```diff
@@ -0,0 +1,244 @@
+
+<hr/>
+
+**Try:** Bring a ``||variables:set [my sprite] to [ ] of kind [player]||`` into the ``||loops:on start||`` block and 
```
using brackets in those inline blocks is a new style; maybe worth considering applying that to our docs in general @ganicke ?

ganicke replied:
Is this what we're going for or is there a new style delimited with ` [ ]` that I'm not seeing?

![image](https://user-images.githubusercontent.com/27789908/101826063-58660400-3ae3-11eb-9bcd-b25b925d624f.png)

kiki-lee replied:
That's it

pelikhan commented on tutorials/interface/activity1.md:
```diff
@@ -0,0 +1,244 @@
+**Try:** Bring a ``||variables:set [my sprite] to [ ] of kind [player]||`` into the ``||loops:on start||`` block and 
+play around with it until your [*sprite*](#sprote "A 2-D image that moves on the screen") shows on the screen.
+
+>>*Tip: Drag the ``||game:splash "___"||`` out of the ``||loops:on start||`` block
```
We have a hint macro that you could potentially use
```
### ~ hint

### ~
```

kiki-lee replied:
I tried that originally, but it didn't do anything in the tutorial-tool, so I didn't know if it was implemented in arcade.

kiki-lee committed:
Update activity1.md

Added the last of the imagery.  Should be good to go!

shakao commented on tutorials/interface/activity1.md:
```diff
@@ -0,0 +1,244 @@
+
+Keep going to learn more about blocks.
+
+![Block Animation](https://raw.githubusercontent.com/microsoft/pxt-arcade/master/docs/static/tutorials/use_blocks.gif "Blocks appear, duplicate, and delete." )
```
```suggestion
![Block Animation](/tutorials/use_blocks.gif "Blocks appear, duplicate, and delete." )

```

shakao commented:
looks good to me! the one other thing you'll want is to add "tutorials/interface/activity1.md" to the pxt.json file here: https://github.com/microsoft/pxt-skillmap-sample/blob/main/pxt.json#L8

kiki-lee committed:
Update tutorials/interface/activity1.md

Co-authored-by: shakao <34112083+shakao@users.noreply.github.com>
